spec/lang/machine.md

```diff
@@ -274,9 +278,18 @@ impl<M: Memory> ThreadManager<M> {
         }
     }
 
-    pub fn spawn(&mut self, func: Function) -> NdResult<ThreadId> {
+    pub fn spawn(&mut self, mem: &mut AtomicMemory<M>, func: Function) -> NdResult<ThreadId> {
```

Comment:
Please just move this function to `impl Machine`, if it needs memory.

Reply:
To be honest I was thinking of just removing the ThreadManager entirely. It does not really have any purpose anymore.

Reply:
Yeah that also works for me.